src/client/components/fetchBtn.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+class FetchBtn extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    const { onClick } = this.props;
+    axios.get('api/tabledatas')
```

Comment:
Instead of handling promise directly, I suggest to leverage `middleware`, [redux-thunk](https://github.com/reduxjs/redux-thunk) or [redux-saga](https://github.com/redux-saga/redux-saga) to handle  asynchronous actions.